spring-kafka/src/main/java/org/springframework/kafka/listener/DeadLetterPublishingRecoverer.java

```diff
@@ -75,6 +75,8 @@ public class DeadLetterPublishingRecoverer implements ConsumerAwareRecordRecover
 
 	private static final long THIRTY = 30L;
 
+	private static final ThreadLocal<ConsumerRecord<?, ?>> IN_RECORD = new ThreadLocal<>();
```

Comment:
May be not `static` for cases when several `DeadLetterPublishingRecoverer` can be used in the project?

Reply:
Yeah... Sure. I realized that later on.
Thanks for confirmation.

Merging...